### PR TITLE
CORTX-31001: Non Jenkins RE Deployment steps to be modified for community-CORTX-Deployment

### PR DESCRIPTION
**Pull request by swatid-seagate**
Signed-off-by: Swatid <swati.dhotarkar@seagate.com>

# Problem Statement
-Non Jenkins RE Deployment steps to be modified for community-CORTX-Deployment
https://jts.seagate.com/browse/CORTX-31001

# Review Checklist 
  Checklist for Author
- [Done ] JIRA number/GitHub Issue added to PR
- [ Done] Jira and state/status is updated and JIRA is updated with PR link
- [Done ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to [solutions/community-deploy/CORTX-Deployment.md](https://github.com/swatid-seagate/cortx-re/blob/cortx-deployment-update/solutions/community-deploy/CORTX-Deployment.md)

-----
[View rendered solutions/community-deploy/CORTX-Deployment.md](https://github.com/swatid-seagate/cortx-re/blob/cortx-deployment-update/solutions/community-deploy/CORTX-Deployment.md)

### CODE REVIEW

File: solutions/community-deploy/CORTX-Deployment.md [hunk 1]
```diff
@@ -1,84 +1,102 @@
...
-### Minimum specification required for Cortx Stack
+Following are the prerequisites to set up the K8s cluster and deploy the CORTX stack
+
+### Minimum system specifications required for Cortx Stack
    - RAM: 16GB
    - CPU: 8Core
    - DISK: Need 8 Disk with 25GB per disk
 
-### There are two types config for `solution.yaml` file manual and automated. If you have automated config type for `solution.yaml`, please make sure your vm having below drives available
+### Types of configurations available for `solution.yaml` file
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:13]: This is not a prerequisite. This should go into a separate section in Deploy the Cortx Stack on K8s Cluster

File: solutions/community-deploy/CORTX-Deployment.md [hunk 2]
```diff
@@ -1,84 +1,102 @@
...
+-  Manual 
+-  Automated: When using automated configuration, please make sure your VM(virtual machine) has following drives available on it:
+
 ```
-      ls /dev/sd*
+ls /dev/sd*
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:18]: Please paste the exact command output for better understanding. e.g. 
```
[root@test ~]# ls /dev/sd*
/dev/sda  /dev/sda1  /dev/sda2  /dev/sdb  /dev/sdc  /dev/sdd  /dev/sde  /dev/sdf  /dev/sdg  /dev/sdh  /dev/sdi
[root@test ~]#
```

File: solutions/community-deploy/CORTX-Deployment.md [hunk 3]
```diff
@@ -1,84 +1,102 @@
...
 
 ```
-      reboot
+reboot
 ```   
+
 ### Note
- 1. All nodes should be reachable over SSH with root user
+ 1. All the nodes should be reachable over SSH with root user.
  2. You also deploy CORTX on AWS EC2 instance. Please follow [CORTX Deployment on AWS](https://github.com/Seagate/cortx-re/blob/main/solutions/community-deploy/cloud/AWS/README.md)
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:49]: You also deploy CORTX on AWS EC2 instance
should be
You can deploy CORTX on AWS EC2 instance also.

File: solutions/community-deploy/CORTX-Deployment.md [hunk 4]
```diff
@@ -1,84 +1,102 @@
...
 
 ## Install K8s cluster
+**To install the K8s cluster, run the following commands:**
 
-   - Clone cortx-re repository and change directory to `cortx-re/solutions/kubernetes`
+Clone cortx-re repository and change directory to `cortx-re/solutions/kubernetes`
 ```
-      git clone https://github.com/Seagate/cortx-re 
-      cd $PWD/cortx-re/solutions/kubernetes
+git clone https://github.com/Seagate/cortx-re
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:56]: Use  `git clone https://github.com/Seagate/cortx-re && cd $PWD/cortx-re/solutions/kubernetes`

File: solutions/community-deploy/CORTX-Deployment.md [hunk 5]
```diff
@@ -1,84 +1,102 @@
...
+Clone cortx-re repository and change directory to `cortx-re/solutions/kubernetes`
 ```
-      git clone https://github.com/Seagate/cortx-re 
-      cd $PWD/cortx-re/solutions/kubernetes
+git clone https://github.com/Seagate/cortx-re 
+cd $PWD/cortx-re/solutions/kubernetes
 ```
-   - Create hosts file by below command, change root user password in below command.
+
+To create the host file and change the root user password, run the following command
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:60]: Please mention that change `rootuserpassword` with your system password.

File: solutions/community-deploy/CORTX-Deployment.md [hunk 6]
```diff
@@ -1,84 +1,102 @@
...
 ```
-      echo "hostname=$(hostname),user=root,pass=rootuserpassword" > hosts && cat hosts
+echo "hostname=$(hostname),user=root,pass=rootuserpassword" > hosts && cat hosts
 ```
-   - Execute `cluster-setup.sh` to setup K8s cluster on your VM. If you wanted to run pods on to master node as well then we need to give first input parameter for `cluster-setup.sh` script true else false. If you have single node cluster then pass input parameter should be true.
+
+Execute `cluster-setup.sh` to setup K8s cluster on your VM.  
+If you want to run pods on to the master node as well, give first input parameter for `cluster-setup.sh` script as true else false.
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:66]: You do not need to mention `false` explicitly.

File: solutions/community-deploy/CORTX-Deployment.md [hunk 7]
```diff
@@ -1,84 +1,102 @@
...
+If you want to run pods on to the master node as well, give first input parameter for `cluster-setup.sh` script as true else false.  
+If you have single node cluster, pass the input parameter as true.
+
 ```
-      ./cluster-setup.sh true
+./cluster-setup.sh true
 ```
 
-## Deploy Cortx Stack on K8s Cluster
+## Deploy the Cortx Stack on K8s Cluster
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:73]: s/Cortx/CORTX

File: solutions/community-deploy/CORTX-Deployment.md [hunk 8]
```diff
@@ -1,84 +1,102 @@
...
-   7. S3_EXTERNAL_HTTP_NODEPORT - If you want to use different port for HTTP Port to IO service, then export this variable with another port (Default value S3_EXTERNAL_HTTP_NODEPORT="30080").
-   8. S3_EXTERNAL_HTTPS_NODEPORT - If you want to use different port for HTTPS Port to IO service, then export this variable with another port (Default value S3_EXTERNAL_HTTPS_NODEPORT="30443").
-   9. SOLUTION_CONFIG_TYPE - There are two types config for solution yaml file manual and automated. Automated means solution.yaml is created automatically but it needs VM with required specifications and manual means we need to create solution.yaml file and replace it at script location. If you wanted to create solution yaml manually, then create it and place it at script location and configure SOLUTION_CONFIG_TYPE variable as manual (Default value SOLUTION_CONFIG_TYPE=manual).
+Note:
+   1. CORTX_SCRIPTS_BRANCH : If you want to use another cortx-K8s branch then export this variable with your branch (Default value CORTX_SCRIPTS_BRANCH="v0.2.1").
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:82]: Maybe we should mention the default values of all the parameters also. We can use below format,
Variable Name |  Default value | description

File: solutions/community-deploy/CORTX-Deployment.md [hunk 9]
```diff
@@ -1,84 +1,102 @@
...
+   6. CONTROL_EXTERNAL_NODEPORT : If you want to use different port for control service, export this variable with another port (Default value CONTROL_EXTERNAL_NODEPORT="31169").
+   7. S3_EXTERNAL_HTTP_NODEPORT : If you want to use different port for HTTP Port to IO service, then export this variable with another port (Default value S3_EXTERNAL_HTTP_NODEPORT="30080").
+   8. S3_EXTERNAL_HTTPS_NODEPORT : If you want to use different port for HTTPS Port to IO service, then export this variable with another port (Default value S3_EXTERNAL_HTTPS_NODEPORT="30443").
+   9. SOLUTION_CONFIG_TYPE : There are two types config for solution.yaml file; manual and automated. 
+   - Automated means solution.yaml is created automatically but it needs VM with required specifications.
+   - Manual means we need to create solution.yaml file and replace it at script location. If you wanted to create solution yaml manually, then create it and place it at script location and configure SOLUTION_CONFIG_TYPE variable as manual (Default value SOLUTION_CONFIG_TYPE=manual).
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:92]: Maybe we need to rephrase this better 
Automated - solution.yaml is created by script if VM is created as per standard specification.
Manual - User needs to create solution.yaml with required disks. image details etc.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: solutions/community-deploy/CORTX-Deployment.md [hunk 10]
```diff
@@ -1,84 +1,107 @@
...
+
+To create the host file and change the root user password with your system password, run the following command
 ```
-      echo "hostname=$(hostname),user=root,pass=rootuserpassword" > hosts && cat hosts
+# echo "hostname=$(hostname),user=root,pass=rootuserpassword" > hosts && cat hosts
 ```
-   - Execute `cluster-setup.sh` to setup K8s cluster on your VM. If you wanted to run pods on to master node as well then we need to give first input parameter for `cluster-setup.sh` script true else false. If you have single node cluster then pass input parameter should be true.
+
+Execute `cluster-setup.sh` to setup K8s cluster on your VM.  
+If you want to run pods on to the master node as well, pass the first input parameter for `cluster-setup.sh` script as true.
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:68]: s/master/primary/

Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:68]: Can we make this bulleted list or make it single line

File: solutions/community-deploy/CORTX-Deployment.md [hunk 11]
```diff
@@ -1,84 +1,107 @@
...
+   4. CORTX_SERVER_IMAGE | CORTX_SERVER_IMAGE=ghcr.io/seagate/cortx-rgw:2.0.0-latest : Also, if you want to use different server image then export this variable with new image.
+   5. CORTX_DATA_IMAGE | CORTX_DATA_IMAGE=ghcr.io/seagate/cortx-data:2.0.0-latest : Also, if you want to use different data image then export this variable with new image.
+   6. CONTROL_EXTERNAL_NODEPORT | CONTROL_EXTERNAL_NODEPORT="31169" : If you want to use different port for control service, export this variable with another port.
+   7. S3_EXTERNAL_HTTP_NODEPORT | S3_EXTERNAL_HTTP_NODEPORT="30080" : If you want to use different port for HTTP Port to IO service, then export this variable with another port.
+   8. S3_EXTERNAL_HTTPS_NODEPORT | S3_EXTERNAL_HTTPS_NODEPORT="30443" : If you want to use different port for HTTPS Port to IO service, then export this variable with another port.
+   9. SOLUTION_CONFIG_TYPE | SOLUTION_CONFIG_TYPE=manual : There are two types config for solution.yaml file; manual and automated. 
+   -  Automated - solution.yaml is created by script if VM is created as per standard specification.
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:96]: Please use SOLUTION_CONFIG_TYPE variable and use the same format as above.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: solutions/community-deploy/CORTX-Deployment.md (hunk 1 above)
Comment (shailesh-vaidya) [reply solutions/community-deploy/CORTX-Deployment.md:15]: @swatid-seagate Could you please address this.

File: solutions/community-deploy/CORTX-Deployment.md (hunk 10 above)
Comment (shailesh-vaidya) [reply solutions/community-deploy/CORTX-Deployment.md]: Please remove word `master`

File: solutions/community-deploy/CORTX-Deployment.md [hunk 12]
```diff
@@ -1,84 +1,106 @@
...
+-  If you want to run pods on to the master/primary node as well, pass the first input parameter for `cluster-setup.sh` script as true.  
+-  If you have single node cluster, pass the input parameter as true.
+
 ```
-      ./cluster-setup.sh true
+# ./cluster-setup.sh true
 ```
 
-## Deploy Cortx Stack on K8s Cluster
+## Deploy the CORTX Stack on K8s Cluster
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/CORTX-Deployment.md:75]: We have the heading `Deploy the CORTX Stack on K8s cluster` at the top. So this should be only `Deploy CORTX Stack`

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please update PR title and fill PR template details.

**Review by shailesh-vaidya — DISMISSED** [review]
Approved. Please fix Codacy check

File: solutions/community-deploy/CORTX-Deployment.md [hunk 13]
```diff
@@ -1,84 +1,104 @@
...
-   8. S3_EXTERNAL_HTTPS_NODEPORT - If you want to use different port for HTTPS Port to IO service, then export this variable with another port (Default value S3_EXTERNAL_HTTPS_NODEPORT="30443").
-   9. SOLUTION_CONFIG_TYPE - There are two types config for solution yaml file manual and automated. Automated means solution.yaml is created automatically but it needs VM with required specifications and manual means we need to create solution.yaml file and replace it at script location. If you wanted to create solution yaml manually, then create it and place it at script location and configure SOLUTION_CONFIG_TYPE variable as manual (Default value SOLUTION_CONFIG_TYPE=manual).
+**Note:**  
+
+Following parameter/s are passed when the cluster deployment command executes. If no parameter is passed, the default ones are chosen.
+( Format used here : parameter name | default parameter value : description )
+
+   1. CORTX_SCRIPTS_BRANCH | CORTX_SCRIPTS_BRANCH="v0.2.1" : If you want to use another cortx-K8s branch then export this variable with your branch.
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/CORTX-Deployment.md:86]: Can we add parameters in table format? refer below as an example.
https://github.com/marketplace/actions/action-git-diff-suggestions#inputs

**Review by shailesh-vaidya — APPROVED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**